packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,345 @@
+     * for single-node container jobs.
+     */
+    readonly instanceType: ec2.InstanceType;
+
+    /**
+     * The IAM role that the container can assume for AWS permissions.
+     *
+     * @default - An IAM role will created
+     */
+    readonly jobRole?: iam.IRole;
+
+    /**
+     * Linux-specific modifications that are applied to the container, such as details for device mappings.
+     * For now, only the `devices` property is supported.
+     *
+     * @default - None will be used
+     */
+    readonly linuxParams?: ecs.LinuxParameters;
+
+    /**
+     * The hard limit (in MiB) of memory to present to the container. If your container attempts to exceed
+     * the memory specified here, the container is killed. You must specify at least 4 MiB of memory for a job.
+     *
+     * @default 4
+     */
+    readonly memoryLimitMiB?: number;
+
+    /**
+     * The mount points for data volumes in your container.
+     *
+     * @default - No mount points will be used
+     */
+    readonly mountPoints?: ecs.MountPoint[];
+
+    /**
+     * When this parameter is true, the container is given elevated privileges on the host container instance (similar to the root user).
+     * @default false
+     */
+    readonly privileged?: boolean;
+
+    /**
+     * When this parameter is true, the container is given read-only access to its root file system.
+     *
+     * @default false
+     */
+    readonly readOnly?: boolean;
+
+    /**
+     * The number of physical GPUs to reserve for the container. The number of GPUs reserved for all
+     * containers in a job should not exceed the number of available GPUs on the compute resource that the job is launched on.
+     *
+     * @default - No GPU reservation
+     */
+    readonly gpuCount?: number;
+
+    /**
+     * A list of ulimits to set in the container.
+     */
+    readonly ulimits: ecs.Ulimit[];
+
+    /**
+     * The user name to use inside the container.
+     *
+     * @default - None will be used
+     */
+    readonly user?: string;
+
+    /**
+     * The number of vCPUs reserved for the container. Each vCPU is equivalent to
+     * 1,024 CPU shares. You must specify at least one vCPU.
+     *
+     * @default 1
+     */
+    readonly vcpus?: number;
+
+    /**
+     * A list of data volumes used in a job.
+     *
+     * @default - No data volumes will be used
+     */
+    readonly volumes?: ecs.Volume[];
+}
+
+/**
+ * Properties for a job definition
+ */
+export interface JobDefinitionProps {
+    /**
+     * The name of the job definition.
+     *
+     * Up to 128 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed.
+     *
+     * @attribute
+     * @default Cloudformation-generated name
+     */
+    readonly jobDefinitionName?: string;
+
+    /**
+     * An object with various properties specific to container-based jobs.
+     *
+     * @default - undefined
+     */
+    readonly containerProps?: IJobDefinitionContainer;
```

Comment:
How about just `container` for this name?

Reply:
Got it 👍